core/Types.h

```diff
@@ -75,13 +80,22 @@ class Types final {
     /** Lower upper bound: the narrowest type that is supper type of both t1 and t2 */
     static TypePtr any(Context ctx, const TypePtr &t1, const TypePtr &t2);
 
-    /** is every instance of  t1 an  instance of t2? */
-    static bool isSubTypeUnderConstraint(Context ctx, TypeConstraint &constr, const TypePtr &t1, const TypePtr &t2);
+    /**
+     * is every instance of  t1 an  instance of t2?
+     *
+     * The parameter `mode` controls whether or not `T.untyped` is
+     * considered to be a super type or subtype of all other types */
+    static bool isSubTypeUnderConstraint(Context ctx, TypeConstraint &constr, UntypedMode mode, const TypePtr &t1,
```

Comment:
the order of arguments was surprising for me(it's different from other functions in this family), but can get used to it

Reply:
Happy to change the order, where would you prefer the `mode` argument go?

Reply:
after `t2`?

Reply:
Happy to move it there :)